plat/nvidia/tegra/common/tegra_bl31_setup.c

```diff
@@ -226,3 +227,35 @@ void bl31_plat_arch_setup(void)
 	/* enable the MMU */
 	enable_mmu_el3(0);
 }
+
+/*******************************************************************************
+ * Check if the given NS DRAM range is valid
+ ******************************************************************************/
+int bl31_check_ns_address(uint64_t base, uint64_t size_in_bytes)
+{
+	uint64_t end = base + size_in_bytes;
+
+	/*
+	 * Check if the NS DRAM address is valid
+	 */
+	if ((base < TEGRA_DRAM_BASE) || (end > TEGRA_DRAM_END) ||
+	    (base == end)) {
+		ERROR("NS address is out-of-bounds!\n");
+		return -EFAULT;
+	}
+
+	/*
+	 * TZDRAM aperture contains the BL31 and BL32 images, so we need
+	 * to check if the NS DRAM range overlaps the TZDRAM aperture.
+	 */
+	if (((base >= tegra_bl31_phys_base) && (end <= TZDRAM_END)) ||
+	    ((base <= tegra_bl31_phys_base) && (end >= TZDRAM_END)) ||
+	    ((base >= tegra_bl31_phys_base) && (base <= TZDRAM_END)) ||
+	    ((end >= tegra_bl31_phys_base) && (end <= TZDRAM_END))) {
```

Comment:
This condition is more complex than required. An overlap is not present if:

    `(base >= TZDRAM_END) || (end <= tegra_bl31_phys_base)`

either the region is entirely after the TZDRAM aperture or entirely before it. In any other condition there must be an overlap. So the overlap condition could be written as

    `((base < TZDRAM_END) && (end > tegra_l31_phys_base))`